core/app/[locale]/layout.tsx

```diff
@@ -3,39 +3,25 @@ import { Analytics } from '@vercel/analytics/react';
 import { SpeedInsights } from '@vercel/speed-insights/next';
 import { clsx } from 'clsx';
 import type { Metadata } from 'next';
-import { DM_Serif_Text, Inter, Roboto_Mono } from 'next/font/google';
+import { draftMode } from 'next/headers';
 import { NextIntlClientProvider } from 'next-intl';
 import { getMessages, setRequestLocale } from 'next-intl/server';
 import { NuqsAdapter } from 'nuqs/adapters/next/app';
 import { PropsWithChildren } from 'react';
 
 import '../globals.css';
 
+import { fonts } from '~/app/fonts';
 import { client } from '~/client';
 import { graphql } from '~/client/graphql';
 import { revalidate } from '~/client/revalidate-target';
+import { SiteTheme } from '~/lib/makeswift/components/site-theme';
+import { MakeswiftProvider } from '~/lib/makeswift/provider';
 
 import { Notifications } from '../notifications';
 import { Providers } from '../providers';
 
-const inter = Inter({
-  display: 'swap',
-  subsets: ['latin'],
-  variable: '--font-family-body',
-});
-
-const dm_serif_text = DM_Serif_Text({
-  display: 'swap',
-  subsets: ['latin'],
-  weight: '400',
-  variable: '--font-family-heading',
-});
-
-const roboto_mono = Roboto_Mono({
-  subsets: ['latin'],
-  display: 'swap',
-  variable: '--font-family-mono',
-});
```

Comment:
This is a part of a [separate standalone PR against `soul/main`](https://github.com/bigcommerce/catalyst/pull/1807), won't be here after the rebase.